src/components/[guild]/AccessHub/hooks/useEditGuildPlatform.ts

```diff
@@ -32,6 +33,30 @@ const useEditGuildPlatform = ({
             if (gp.id === guildPlatformId) return response
             return gp
           }),
+          roles: CAPACITY_TIME_PLATFORMS.includes(
+            PlatformType[response.platformId] as PlatformName
+          )
+            ? prevGuild.roles.map((role) => {
```

Comment:
Altough it's only used in two places, I think we should abstract it into a util function (not necessarily in the utils folder, just in one of the files and export it from there), it's complex enough and would be more self-explaining too if we label it somehow